paddle/phi/core/ddim.cc

```diff
@@ -171,11 +171,25 @@ DDim stride_numel(const DDim& ddim) {
   return strides;
 }
 
-DDim DDim::reshape(const std::vector<int>& shape) const {
+DDim DDim::reshape(const std::vector<int>& new_shape) const {
+  std::vector<int> shape = new_shape;
```

Comment:
Creating a new vector here and do ifelse branch in core/ddim.cc will influence each cpu/gpu op I think ?